uid/snowid_test.go

```diff
@@ -0,0 +1,342 @@
+package uid
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"math"
+	"math/rand"
+	"strings"
+	"sync"
+	"testing"
+
+	"golang.org/x/sync/errgroup"
+	"gotest.tools/v3/assert"
+)
+
+func TestNewNode(t *testing.T) {
+	_, err := NewNode(0)
+	assert.NilError(t, err)
+
+	_, err = NewNode(5000)
+	assert.ErrorContains(t, err, "number must be between")
+}
+
+// lazy check if Generate will create duplicate IDs
+// would be good to later enhance this with more smarts
+func TestGenerateDuplicateID(t *testing.T) {
+	node, _ := NewNode(1)
+
+	var x, y ID
+	for i := 0; i < 1000000; i++ {
+		y = node.Generate()
+		if x == y {
+			t.Errorf("x(%d) & y(%d) are the same", x, y)
+		}
+		x = y
+	}
+}
+
+func TestNode_Generate_ConcurrencyAndUniqueness(t *testing.T) {
```

Comment:
This is a new property test, to show that IDs are unique, and to catch any concurrency bugs in `NewNode`.